File: src/Organization.Api/Entities/OrgUnit.cs
using StarterKit.Shared.Entities;

namespace StarterKit.Organization.Api.Entities;

public class OrgUnit : AuditableEntity
{
    public string CompanyId { get; set; } = null!;

    public string? ParentId { get; set; }

    public OrgUnit? Parent { get; set; }

    public IList<OrgUnit> Children { get; set; } = [];

    public OrgUnitType Type { get; set; }

    public string Name { get; set; } = null!;

    public string Code { get; set; } = null!;

    public string? ManagerEmployeeId { get; set; }

    public string? Description { get; set; }

    public OrganizationStatus Status { get; set; } = OrganizationStatus.Active;

    public IList<EmployeeOrgUnitMembership> Memberships { get; set; } = [];
}

File: clients/admin/src/features/departments/types/org-unit.ts
/** Mirrors `Organization.Contracts/OrgUnits/OrgUnitType.cs` — serialized by name. */
export enum OrgUnitType {
  Department = "Department",
  Team = "Team",
}

export interface OrgUnitDto {
  id: string;
  companyId: string;
  parentId?: string | null;
  type: OrgUnitType;
  name: string;
  code: string;
  managerEmployeeId?: string | null;
  description?: string | null;
  status: string;
}

export interface CreateOrgUnitRequest {
  companyId: string;
  parentId?: string;
  type: OrgUnitType;
  name: string;
  code: string;
  managerEmployeeId?: string;
  description?: string;
}

export interface MoveOrgUnitRequest {
  newParentId?: string;
}

export interface OrgUnitTreeNodeDto {
  id: string;
  parentId?: string | null;
  type: OrgUnitType;
  name: string;
  code: string;
  managerEmployeeId?: string | null;
  status: string;
  children: OrgUnitTreeNodeDto[];
}

/** Flattens a tree into a plain list — used to populate org-unit pickers (e.g. the "assign employee" dialog). */
export function flattenOrgUnitTree(nodes: OrgUnitTreeNodeDto[]): OrgUnitTreeNodeDto[] {
  const result: OrgUnitTreeNodeDto[] = [];
  for (const node of nodes) {
    result.push(node);
    if (node.children.length > 0) result.push(...flattenOrgUnitTree(node.children));
  }
  return result;
}
